include/deal.II/base/tensor_base.h

```diff
@@ -359,70 +379,84 @@ class Tensor<1,dim,Number>
   typedef typename numbers::NumberTraits<Number>::real_type real_type;
 
   /**
-   * Declare an array type which can be used to initialize statically an
-   * object of this type.
-   *
-   * Avoid warning about zero-sized array for <tt>dim==0</tt> by choosing
-   * lunatic value that is likely to overflow memory limits.
+   * The tensor type this object represents.
+   */
+  typedef Tensor<rank_,dim,Number> tensor_type;
+
+  /**
+   * Type of stored objects (i.e., the object returned by operator[]()).
+   * This is a tensor of lower rank.
+   */
+  typedef typename Tensor<rank_-1,dim,Number>::tensor_type value_type;
```

Comment:
No, it is not. ```value_type``` is the return type of ```operator[]()``` - at least we use it like that. :-)

If we clean up the mechanism with ```tensor_type``` described below, then we will not have to discuss anything anymore :-).